lib/Logger/Logger.cpp

```diff
@@ -325,88 +325,278 @@ void Logger::log(char const* function, char const* file, int line,
+      buf.appendInteger(uint64_t(_cachedPid));
+    }
 
-  // check if we must display the line number
-  if (_showLineNumber && file != nullptr && function != nullptr) {
-    char const* filename = file;
+    // tid
+    if (_showThreadIdentifier) {
+      static char const s[] = "tid";
 
-    if (_shortenFilenames) {
-      // shorten file names from `/home/.../file.cpp` to just `file.cpp`
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+      buf.appendInteger(uint64_t(Thread::currentThreadNumber()));
+    }
+
+    // thread name
+    if (_showThreadName) {
+      static char const s[] = "thread";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+
+      char const* threadName = Thread::currentThreadName();
+      if (threadName == nullptr) {
+        threadName = "main";
+      }
+
+      buf.appendJsonEncoded(threadName, strlen(threadName));
+    }
+
+    // role
+    if (_showRole) {
+      static char const s[] = "role";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+
+      if (_role != '\0') {
+        buf.appendJsonEncoded(&_role, 1);
+      } else {
+        buf.appendText("\"\"", 2);
+      }
+    }
+
+    // log level
+    {
+      static char const s[] = "level";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+      auto const& ll = Logger::translateLogLevel(level);
+      buf.appendJsonEncoded(ll.c_str(), ll.size());
+    }
+
+    // file and line
+    if (_showLineNumber && file != nullptr) {
+      static char const s[] = "file";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+
+      char const* filename = file;
       char const* shortened = strrchr(filename, TRI_DIR_SEPARATOR_CHAR);
       if (shortened != nullptr) {
         filename = shortened + 1;
       }
+
+      buf.appendJsonEncoded(filename, strlen(filename));
     }
+
+    if (_showLineNumber && file != nullptr) {
+      static char const s[] = "line";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+      buf.appendInteger(uint64_t(line));
+    }
+
+    if (_showLineNumber && function != nullptr) {
+      static char const s[] = "function";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+      buf.appendJsonEncoded(function, strlen(function));
+    }
+
+    // the topic
+    {
+      static char const s[] = "topic";
+
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+
+      std::string topic = LogTopic::lookup(topicId);
+      buf.appendJsonEncoded(topic.c_str(), topic.size());
+    }
+
+    // the logid
+    if (::arangodb::Logger::getShowIds()) {
+      static char const s[] = "lid";
```

Comment:
I suggest we name this `id` and not `lid`.

Reply:
ok

Reply:
Done